Update inotify to 0.10

fornwall opened the pull request:
See the [CHANGELOG](https://github.com/hannobraun/inotify-rs/blob/main/CHANGELOG.md).

fornwall commented:
Hi @0xpr03, could you have a look at this PR?

0xpr03 commented on notify/src/inotify.rs:
```diff
@@ -307,9 +307,11 @@ impl EventLoop {
                                     }
                                     None => {
-                                        log::trace!("No patch for DELETE_SELF event, may be a bug?");
+                                        log::trace!(
```
was this part of cargo fmt ?

fornwall replied:
Yes, it was.

0xpr03 commented:
Is there a strong reason to push for 0.10 ASAP ? I don't see any real reasons in their changelog.

0xpr03 commented on notify/src/inotify.rs:
```diff
@@ -471,7 +474,8 @@ impl EventLoop {
                             if p.starts_with(&path) {
                                 inotify
-                                    .rm_watch(w.clone())
+                                    .watches()
```
The watches() call seems to perform an Arc clone and creating some sort of Struct containing a [FdGuard](https://github.com/hannobraun/inotify-rs/blob/39b64297a6d3dd2706652c0e19d061287a4e04be/src/fd_guard.rs#L22). It might be in our interest to use one watches() call for at least the whole loop.

fornwall replied:
Thanks, will look into this.

fornwall replied:
Done in b70b72b79f9ad273a2f36a3d435a97f069b360ca

0xpr03 reviewed (changes requested):
Blocked by #549.
And we might want to look into re-using watches() for hot loops.

0xpr03 commented:
Thanks, I've added one comment regarding the watches() overhead.

fornwall commented:
> Is there a strong reason to push for 0.10 ASAP ? I don't see any real reasons in their changelog.

The main driver here is to avoid duplication - other crates in the ecosystem are using 0.10 (for instance https://gitlab.com/gilrs-project/gilrs), and there is a desire (or rule) in many projects to avoid pulling in multiple versions of the same crate.

So in general, unless there is a good reason, I think the latest released version of a crate should be used.

fornwall commented:
Tested this a bit, and it seems we need to adopt to: https://github.com/hannobraun/inotify-rs/pull/190

fornwall commented:
> Thanks, I've added one comment regarding the watches() overhead.

Thank you for the review! Addressed that one in b70b72b79f9ad273a2f36a3d435a97f069b360ca

MarijnS95 commented:
> The main driver here is to avoid duplication - other crates in the ecosystem are using 0.10 (for instance https://gitlab.com/gilrs-project/gilrs), and there is a desire (or rule) in many projects to avoid pulling in multiple versions of the same crate.

Yeah, same for us, we'd love to see both crates in sync :)

0xpr03 commented:
Thanks, I got to merge the blocker and fixed one small readme formatting error while rebasing for (valid) CI results.

0xpr03 commented:
Now let's see, there is some hope I can get around to a new release this weekend, but no promises.
